workspaces/ui-v2/src/pages/diffs/EndpointDocumentationPane.tsx

```diff
@@ -110,65 +111,91 @@ export const EndpointDocumentationPane: FC<
-                    />
-                  )}
-                </ShapeFetcher>
-              </div>
+      {thisEndpoint.requests.length > 0 && (
+        <HighlightedLocation
+          className={classes.bodyContainer}
+          targetLocation={highlightedLocation}
+          expectedLocation={Location.Request}
+        >
+          <div id="request-body">
+            <h6 className={classes.bodyHeader}>Request Body</h6>
+            <div className={classes.bodyDetails}>
+              <HttpBodySelector
+                items={thisEndpoint.requests}
+                getDisplayName={(request) =>
+                  request.body?.contentType || 'No Body'
+                }
+              >
+                {(request) =>
+                  request.body ? (
+                    <ShapeFetcher
+                      rootShapeId={request.body.rootShapeId}
+                      changesSinceBatchCommit={lastBatchCommit}
+                    >
+                      {(shapes) => (
+                        <HttpBodyPanel
+                          shapes={shapes}
+                          // Typescript cannot infer through render props for some reason
+                          location={`Request Body ${request.body!.contentType}`}
+                        />
+                      )}
+                    </ShapeFetcher>
+                  ) : (
+                    <div>No Body Request</div>
+                  )
+                }
+              </HttpBodySelector>
             </div>
-          </HighlightedLocation>
-        </React.Fragment>
-      ))}
-      {thisEndpoint.responseBodies.map((responseBody) => {
-        return (
-          <React.Fragment key={responseBody.responseId}>
-            <HighlightedLocation
-              className={classes.bodyContainer}
-              targetLocation={highlightedLocation}
-              contentType={responseBody.contentType}
-              statusCode={responseBody.statusCode}
-              expectedLocation={Location.Response}
-            >
-              <div id={responseBody.responseId}>
-                <h6 className={classes.bodyHeader}>
-                  {responseBody.statusCode} response
-                </h6>
-                <div className={classes.bodyDetails}>
-                  <ShapeFetcher
-                    rootShapeId={responseBody.rootShapeId}
-                    changesSinceBatchCommit={lastBatchCommit}
-                  >
-                    {(shapes) => (
-                      <HttpBodyPanel
-                        shapes={shapes}
-                        location={`${responseBody.statusCode} response ${responseBody.contentType}`}
-                      />
-                    )}
-                  </ShapeFetcher>
+          </div>
+        </HighlightedLocation>
+      )}
+
+      {selectors
+        .getResponsesInSortedOrder(thisEndpoint.responsesByStatusCode)
+        .map(([statusCode, responses]) => {
```

Comment:
Had to restructure this, such that statusCodes are grouped together. Otherwise we'd have:
200 Response
responseId 1
200 Response
responseId 2

Which isn't easy to read (likely we'd want to group these together)